io.c

```diff
@@ -11624,56 +11585,56 @@ io_s_write(int argc, VALUE *argv, VALUE klass, int binary)
 
 /*
  *  call-seq:
- *     IO.write(name, string [, offset])           -> integer
- *     IO.write(name, string [, offset] [, opt])   -> integer
- *     File.write(name, string [, offset])         -> integer
- *     File.write(name, string [, offset] [, opt]) -> integer
+ *    IO.write(command, string)             -> integer
+ *    IO.write(path, string, offset = 0)    -> integer
  *
- *  Opens the file, optionally seeks to the given <i>offset</i>, writes
- *  <i>string</i>, then returns the length written.  #write ensures the
- *  file is closed before returning.  If <i>offset</i> is not given in
- *  write mode, the file is truncated.  Otherwise, it is not truncated.
+ *  Opens the stream, writes the given +string+ to it,
+ *  and closes the stream; returns the number of bytes written.
  *
- *  If +name+ starts with a pipe character (<code>"|"</code>) and the receiver
- *  is the IO class, a subprocess is created in the same way as Kernel#open,
- *  and its output is printed to the standard output.
- *  Consider to use File.write to disable the behavior of subprocess invocation.
+ *  The first argument must be a string;
+ *  its meaning depends on whether it starts with the pipe character (<tt>'|'</tt>):
  *
- *    File.write("testfile", "0123456789", 20)  #=> 10
- *    # File could contain:  "This is line one\nThi0123456789two\nThis is line three\nAnd so on...\n"
- *    File.write("testfile", "0123456789")      #=> 10
- *    # File would now read: "0123456789"
- *    IO.write("|tr a-z A-Z", "abc")            #=> 3
- *    # Prints "ABC" to the standard output
+ *  - If so (and if +self+ is an instance of \IO),
+ *    the rest of the string is a command to be executed as a subprocess.
+ *  - Otherwise, the string is the path to a file.
+ *
+ *  With argument +command+ given, executes the command in a shell,
+ *  writes its output to $stdout, and returns the length of the given +string+:
+ *
+ *    IO.write('| cat t.txt', 'abd') # => 3
+ *
+ *  Output:
+ *
+ *    First line
+ *    Second line
  *
- *  If the last argument is a hash, it specifies options for the internal
- *  open().  It accepts the following keys:
+ *    Third line
+ *    Fourth line
```

Comment:
The confusion is in the git differ.  t.txt has five lines, the third of which is blank;  this is to support the empty-string line separator that returns 'paragraphs', not lines.

Reply:
Sorry, I'm still a bit confused. The example before this is:

```ruby
IO.write('| cat t.txt', 'abd') # => 3
```

The example after this is:

```ruby
IO.write('t.tmp', 'abc')    # => 3
File.read('t.tmp')          # => "abc"
```

I'm not sure how any of these example involves this output.

Maybe this was supposed to be part of the documentation for `IO.copy_stream`?

Reply:
I don't know what you mean.  The first example has a command that copies t.txt to $stdout.  What I have labelled "Output" is what was copied from the file to $stdout.

Reply:
Ah I see. I was really confused. I think you mean that `t.txt` has the contents

```
First line
Second line

Third line
Fourth line
```

It's really not clear to me that it's the contents of `t.txt`. Also it's really confusing since the command is completely ignoring the standard input.

Maybe something like this would be less confusing (the dash for the filename reads from stdin)?

```
 *    IO.write('| cat t.txt -', 'abc') # => 3
 *
 *  Output (the third line is blank):
 *
 *    First line
 *    Second line
 *
 *    Third line
 *    Fourth line
 *    abc
```

Reply:
Is it important that we output `t.txt`? We're just demonstrating that we can call a shell command and pass stdin into it right? So can we simplify the example?

```
 *    IO.write('| cat', 'Hello World!') # => 12
 *
 *  Output:
 *
 *    Hello World!

Reply:
Another try.

Reply:
Excellent point.  I was just in a t.txt state of mind.